grammar.js

```diff
@@ -250,6 +250,9 @@ module.exports = grammar({
       $._expression,
       $.type_declaration,
     ],
+    [
+      $.empty_parens, $.top_level_fun_def,
+    ],
```

Comment:
This is setting the precedence of one rule over another, so order should be reversed. I also think it's good practice to document the reason for all of these.
```suggestion
    // Ensure `fun foo() end` parses `()` as empty params instead of a nil body.
    [
      $.top_level_fun_def,
      $.empty_parens,
    ],
```